plugins/resources/nomiclabs.utils.js

```diff
@@ -134,6 +134,7 @@ function configureNetworkEnv(env, networkName, networkConfig, provider, isHardha
   env.config.defaultNetwork = networkName;
 
   env.network = {
```

Comment:
The fact that it still create a new object will cause issue with plugin that modify the network object before this is called.
It is better to simply modify the network object

Reply:
If they modify it before, it shouldn't be an issue since we first set all previous object values then only override the needed ones `...env.network`. I'm using the fork with this fork and seems to be working well with hardhat deploy :)

How would you suggest this be implemented? Glad to iterate on it or take a PR heh

Reply:
Plugins could have copied the reference to hre.network before and will later modify an unrelated object. Modifying the network object would be safer.  I proposed this in that comment: https://github.com/wighawag/hardhat-deploy/issues/132#issuecomment-875365555

Reply:
Object.assign retains the reference to the original object too. I think that is more conventional. But simplicity is always good so maybe @wighawag solution of just doing inplace modifications (`env.network.name = networkName` etc.) is better.

Reply:
Object.assign retains the reference to the original object too. I think that is more conventional. But simplicity is always good so maybe @wighawag solution of just doing inplace modifications (env.network.name = networkName etc.) is better.